presto-iceberg/src/test/java/com/facebook/presto/iceberg/procedure/TestExpireSnapshotProcedure.java

```diff
@@ -0,0 +1,295 @@
+            assertUpdate(session, "INSERT INTO " + tableName + " VALUES(2, 'b')", 1);
+            assertUpdate(session, "INSERT INTO " + tableName + " VALUES(3, 'c')", 1);
+            table.refresh();
+            long snapshotId2 = table.currentSnapshot().snapshotId();
+            assertUpdate(session, "INSERT INTO " + tableName + " VALUES(4, 'd')", 1);
+
+            table.refresh();
+            assertHasSize(table.snapshots(), 4);
+
+            // Drop the indicated 2 snapshots
+            assertUpdate(session, format("CALL system.expire_snapshots(table_name => '%s', schema => '%s', snapshot_ids => %s)",
+                    tableName, TEST_SCHEMA, "array[" + snapshotId1 + ", " + snapshotId2 + "]"));
+            table.refresh();
+            assertHasSize(table.snapshots(), 2);
+        }
+        finally {
+            dropTable(tableName);
+        }
+    }
+
+    @Test
+    public void testInvalidExpireSnapshotsCases()
+    {
+        assertQueryFails("CALL system.expire_snapshots('n', table_name => 't')", ".*Named and positional arguments cannot be mixed");
+        assertQueryFails("CALL custom.expire_snapshots('n', 't')", "Procedure not registered: custom.expire_snapshots");
+        assertQueryFails("CALL system.expire_snapshots()", ".*Required procedure argument 'schema' is missing");
+        assertQueryFails("CALL system.expire_snapshots('s', 'n', 2.2)", ".*Cannot cast type decimal\\(2,1\\) to timestamp");
+        assertQueryFails("CALL system.expire_snapshots('', '')", "schemaName is empty");
+    }
+
+    private String getTimestampString(long timeMillsUtc, String zoneId)
+    {
+        Instant instant = Instant.ofEpochMilli(timeMillsUtc);
+        LocalDateTime localDateTime = instant
+                .atZone(ZoneId.of(zoneId))
+                .toLocalDateTime();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
+        formatter = formatter.withZone(ZoneId.of(zoneId));
+        return localDateTime.format(formatter);
+    }
+
+    private Table loadTable(String tableName)
+    {
+        Catalog catalog = CatalogUtil.loadCatalog(HadoopCatalog.class.getName(), ICEBERG_CATALOG, getProperties(), new Configuration());
+        return catalog.loadTable(TableIdentifier.of(TEST_SCHEMA, tableName));
+    }
+
+    private Map<String, String> getProperties()
+    {
+        File metastoreDir = getCatalogDirectory();
+        return ImmutableMap.of("warehouse", metastoreDir.toString());
+    }
+
+    private File getCatalogDirectory()
+    {
+        Path dataDirectory = getDistributedQueryRunner().getCoordinator().getDataDirectory();
+        return dataDirectory.toFile();
+    }
+
+    private long waitUntilAfter(long timestampMillis)
+    {
+        long current = System.currentTimeMillis();
+        if (timestampMillis - current > 10) {
+            throw new PrestoException(StandardErrorCode.INVALID_ARGUMENTS, "Do not allow to spinning wait for longer than 10ms");
+        }
```

Comment:
Perhaps rename argument to be `snapshotTimeMillis`, and call current time `currentTimeMillis`.

We can just use a simple assert for this.

```suggestion
        assertTrue(snapshotTimeMillis - currentTimeMillis <= 10, format("Snapshot time %s is greater than the current time %s by more than 10ms"));
```

Reply:
Sure, thanks for the suggestion, fixed!